chain/chain/src/chain.rs

```diff
@@ -938,6 +941,85 @@ impl Chain {
         }
     }
 
+    /// Check if any orphan with missing chunks is ready to request chunks
+    pub fn check_orphans_with_missing_chunks<F>(
```

Comment:
let's add a test for this function specifically:
 e.g. create a chain with some orphans and call this function to check it finds right things ideally hitting 5 possible conditions.